src/core/public/chrome/ui/header/collapsible_nav_groups.tsx

```diff
@@ -0,0 +1,149 @@
+      navigateToApp,
+      onClick: (event) => {
+        onNavItemClick(event, link);
+      },
+    });
+
+    return {
+      id: `${link.id}-${link.title}`,
+      name: <EuiText size="xs">{link.title}</EuiText>,
+      onClick: euiListItem.onClick,
+      href: euiListItem.href,
+      emphasize: euiListItem.isActive,
+      className: `nav-link-item ${className || ''}`,
+      buttonClassName: 'nav-link-item-btn',
+      'data-test-subj': euiListItem['data-test-subj'],
+      'aria-label': link.title,
+    };
+  };
+  const createSideNavItem = (
+    navLink: LinkItem,
+    level: number,
+    className?: string
+  ): EuiSideNavItemType<{}> => {
+    if (navLink.itemType === LinkItemType.LINK) {
+      if (navLink.link.title === titleForSeeAll) {
+        const navItem = createNavItem({
+          link: navLink.link,
+        });
+
+        return {
+          ...navItem,
+          name: <EuiText color="success">{navItem.name}</EuiText>,
+          emphasize: false,
+        };
+      }
+
+      return createNavItem({
+        link: navLink.link,
+        className,
+      });
+    }
+
+    if (navLink.itemType === LinkItemType.PARENT_LINK && navLink.link) {
+      const props = createNavItem({ link: navLink.link });
+      const parentItem = {
+        ...props,
+        forceOpen: true,
+        /**
+         * The href and onClick should both be undefined to make parent item rendered as accordion.
+         */
+        href: undefined,
+        onClick: undefined,
+        className: classNames(props.className, 'nav-link-parent-item'),
+        buttonClassName: classNames(props.buttonClassName, 'nav-link-parent-item-button'),
+        items: navLink.links.map((subNavLink) =>
+          createSideNavItem(subNavLink, level + 1, 'nav-nested-item')
+        ),
+      };
+      /**
+       * OuiSideBar will never render items of first level as accordion,
+       * in order to display accordion, we need to render a fake parent item.
+       */
+      if (level === LEVEL_FOR_ROOT_ITEMS) {
+        return {
+          className: 'nav-link-fake-item',
+          buttonClassName: 'nav-link-fake-item-button',
+          name: '',
+          items: [parentItem],
+          id: `fake_${props.id}`,
+        };
+      }
+
+      return parentItem;
+    }
+
+    if (navLink.itemType === LinkItemType.CATEGORY) {
+      return {
+        id: navLink.category?.id ?? '',
+        name: <div className="nav-link-item">{navLink.category?.label ?? ''}</div>,
+        items: navLink.links?.map((link) => createSideNavItem(link, level + 1)),
+        'aria-label': navLink.category?.label,
+      };
+    }
+
+    return {} as EuiSideNavItemType<{}>;
+  };
+  const orderedLinksOrCategories = getOrderedLinksOrCategories(navLinks);
+  const sideNavItems = orderedLinksOrCategories
+    .map((navLink) => createSideNavItem(navLink, LEVEL_FOR_ROOT_ITEMS))
+    .filter((item): item is EuiSideNavItemType<{}> => !!item);
```

Comment:
What's the intention of filtering with `!!item`? I see `createSideNavItem` might return `{}`, are you trying to filter out `{}`?

Reply:
Updated.